public/app/plugins/datasource/loki/querySplitting.ts

```diff
@@ -126,6 +126,10 @@ export function runPartitionedQuery(datasource: LokiDatasource, request: DataQue
       .pipe(
         // in case of an empty query, this is somehow run twice. `share()` is no workaround here as the observable is generated from `of()`.
         map((partialResponse) => {
+          if (partialResponse.error) {
+            subscriber.error(partialResponse.error);
+            return mergedResponse || { data: [] };
+          }
```

Comment:
neat. this solves the problem. but... i wonder if we should do this app-logic in the `.map`.

taking one step back, do we really need both the `.map()` and the `.subscribe()` below?

i think we can just move the whole body of `map()` into the `next()`, and we can remove `map()` completely. WDYT?

Reply:
I'll give it a try. Honestly, I only kept it because it was in the original code.

Reply:
Gave it a try and it produced unexpected results.
The bottom line of why I kept it structured that way is that I wanted to split the work into a "map reduce" way, where we 1) do the request 2) transform it 3) report it (in next()).

Reply:
i see your point.
there's still one problem unfortunately, which is, if we do `subscriber.error`, we are not supposed to do `subscriber.next` or `subscriber.complete` ever again. `error` means it's done, and no new data will come.

i guess we could add the error to `mergedResponse.error` ? 

( https://reactivex.io/documentation/contract.html )

Reply:
Ended up moving the code to `next()` and `complete()`, good catch with that!

Reply:
Sorry, missed your response. 

> if we do subscriber.error, we are not supposed to do subscriber.next or subscriber.complete ever again

Oh okay, gotcha.

> i guess we could add the error to mergedResponse.error ?

I'll give it a try 🙇

Reply:
Done. Let me know if you see anything else.